controller_interface/include/semantic_components/led_rgb_device.hpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2024, Sherpa Mobile Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SEMANTIC_COMPONENTS__LED_RGB_DEVICE_HPP_
+#define SEMANTIC_COMPONENTS__LED_RGB_DEVICE_HPP_
+
+#include <string>
+#include <vector>
+
+#include "semantic_components/semantic_component_command_interface.hpp"
+#include "std_msgs/msg/color_rgba.hpp"
+
+namespace semantic_components
+{
+class LedRgbDevice : public SemanticComponentCommandInterface<std_msgs::msg::ColorRGBA>
+{
+public:
+  /**
+   * Constructor for a LED RGB device with interface names set based on device name.
+   * The constructor sets the command interface names to "<name>/r", "<name>/g", "<name>/b".
+   */
+  explicit LedRgbDevice(const std::string & name)
+  : SemanticComponentCommandInterface(
+      name, {{name + "/" + "r"}, {name + "/" + "g"}, {name + "/" + "b"}})
+  {
+  }
+
+  /**
+   * Constructor for a LED RGB device with custom interface names.
+   * The constructor takes the three command interface names for the red, green, and blue channels.
+   */
+  explicit LedRgbDevice(
+    const std::string & interface_r, const std::string & interface_g,
+    const std::string & interface_b)
+  : SemanticComponentCommandInterface("", {{interface_r}, {interface_g}, {interface_b}})
```

Comment:
```suggestion
  : SemanticComponentCommandInterface(interface_r.substr(0, interface_r.find("/")), {{interface_r}, {interface_g}, {interface_b}})
```

In the constructor, it would be great to check if the first sub string of all these 3 interfaces is the same one. Just not to allow controlling interfaces of different device

Reply:
Are you sure we want to fail in the constructor?

In that case, we could force the `name` variable to be the prefix:

```cpp
explicit LedRgbDevice(
    const std::string & name, const std::string & interface_r,
    const std::string & interface_g, const std::string & interface_b)
  : SemanticComponentCommandInterface(name,
    {{name + "/" + interface_r}, {name + "/" + interface_g}, {name + "/" + interface_b}})
```

Reply:
Sure. This makes more sense. Let's go for it

Reply:
https://github.com/ros-controls/ros2_control/pull/1945/commits/e7b078e27e3a60aa9e5d610964df7484fe5046d0